stdlib/src/memory/arc.mojo

```diff
@@ -126,11 +335,12 @@ struct Arc[T: Movable](CollectionElement, CollectionElementNew):
         """Delete the smart pointer reference.
 
         Decrement the ref count for the reference. If there are no more
-        references, delete the object and free its memory."""
-        if self._inner[].drop_ref():
-            # Call inner destructor, then free the memory.
+        references, free its memory."""
+        var should_del = self._inner[].drop_ref()
+        if should_del:
             (self._inner).destroy_pointee()
             self._inner.free()
+        self._inner = UnsafePointer[Self._inner_type]()
```

Comment:
**Request** This "nulling out" of fields is not required within `__del__()` methods. After the `__del__()` completes, this object instance is no longer alive, so there's no code that could observe `self._inner` one way or the other 🙂